### ISSUE TITLE
Fix the labeling of the menu Actions button

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#59431]
### Description

Go to Site Editor, edit mode, select a navigation menu, observe the Settings sidebar:

Screenshot:

![Screenshot 2024-02-28 at 12 23 27](https://github.com/WordPress/gutenberg/assets/1682452/19b70a8c-e320-4915-a14e-9b4da61dbc97)

I have no words to describe this issue, as it reveals a total lack of understanding of what labeling controls is about.

That button gives access to a menu where users can perform some actions:
- Choose a menu
- import a classic menu
- Create a new menu

The button should explain what it gives access to.
Instead, it'a labeled either with the name of the currently selected menu or with `Loading...`.

Aside:
The ellipsis icon:

![Screenshot 2024-02-28 at 12 45 36](https://github.com/WordPress/gutenberg/assets/1682452/2e711fdd-d3d3-419c-ad2b-a5419759cc58)

is pretty universally associated with the concept of 'More'. It should be used and labeled accordingly. It may be complemented with more context e.g. 'More actions' or 'More options' and such. In the editor, there's a large inconsistency with the concept of 'more' and with the way this icon button is used. That should be broadly discussed in a separate issue. For sure, it shouldn't be labeled with the name of the currently selected menu or to communicate the 'loading' state.

### Step-by-step reproduction instructions

See above.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Overall, I think this UI has room for further improvements.
Specifically, the information about the currently selected menu is not visible unless users hover or focus the ellipsis button.
The heading of the panel is always 'Menu' and never changes.
The information about the currently selected menu name should be provided with visible text, either as suffix of the panel heading or below the heading.

**Comment by afercia** [comment]
Working on this surfaced another bug: after switching to another menu, the reported state used for the label is always `Loading…`. The logic to determine this info needs to be revised. Screenshot after witching to another menu:

![Screenshot 2024-02-28 at 15 04 36](https://github.com/WordPress/gutenberg/assets/1682452/f8d91a7b-830e-4ae0-aa86-d0ee8b8dc0b9)

**Comment by helloakshat** [comment]
![more option](https://github.com/WordPress/gutenberg/assets/41269766/26f5ad17-259e-4b72-9868-173a10d7c762)

Here is what I feel like!
The "kebab menu" consists of different types of menus which are categorized differently!
1. View
2. Editor
3. Tools
4. Preferences

So we can't name this option generic like "Preferences, Settings, etc"

Here are the label names that I want to suggest.
1. More Options
2. Advance Option
3. View More

I hope this works.
Thanks

**Comment by unprintedch** [comment]
This icon is consistent with all the other block option. 
It feels pretty clear to me that it leads to more action "in the context of the current block".
Why do you think it should be more explanatory here than in other block?
See capture
<img width="286" alt="Capture d’écran 2024-03-07 à 22 33 42" src="https://github.com/WordPress/gutenberg/assets/2622001/b35e42c8-e4c3-487d-929e-97d73f1a44cc">

**Comment by afercia** [comment]
Looking back into this, there's one more case to report: when a navigation menu has been deleted or is anyways unavailable, the ellipsis button may be totally unlabeled. One  more case of unlabeled control. though it seems an edge case. Turns out [this logic](https://github.com/WordPress/gutenberg/blob/94b8ac4929e7af8f4a4e4b108873d8ac24a688df/packages/block-library/src/navigation/edit/navigation-menu-selector.js#L107-L120) may faiil in some cases when a menu is not available but there is a `currentMenuId` set. I guess this neds to be investigated.

Screenshot of the unlabeled button:

![Image](https://github.com/user-attachments/assets/3f95d73e-e537-43fe-a605-dd0589fd0c91)
